experimental/auth-logic/wrappers/rekor_wrapper.go

```diff
@@ -0,0 +1,131 @@
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"github.com/go-openapi/runtime"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sigstore/rekor/pkg/types"
+	rekord "github.com/sigstore/rekor/pkg/types/rekord/v0.0.1"
+	"io/ioutil"
+)
+
+// This wrapper is meant to be modeled after the comments here
+// https://github.com/project-oak/oak/blob/main/oak_functions/client/rust/src/rekor.rs
+// describing that verifying the log entry entails:
+//  -- verifying the signature in `signedEntryTimestamp`, using Rekor's public key,
+// -- verifying the signature in `body.RekordObj.signature`, using Oak's public key,
+// -- verifying that the content of the body matches the input `endorsement_bytes`.
+// And also possibly validating the inclusion proof
+//
+// TODO look into validating the inclusion proof with
+// logEntryAnon.Verification.ValidateInclusionProof.
+// I'm not sure what to pass for "formats" yet
+// https://github.com/sigstore/rekor/blob/e0b79164f2279ad7c3e723a3ee16afbfcf271188/pkg/generated/models/log_entry.go#L365
+
+func getLogEntryAnonFromFile(rekorLogFilePath string) (*models.LogEntryAnon, error) {
+	// get LogEntry, which is a map from strings to LogEntryAnons
+	logEntryBytes, err := ioutil.ReadFile(rekorLogFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("could not read the rekor log file: %v", err)
+	}
+
+	var logEntry models.LogEntry
+
+	err = json.Unmarshal(logEntryBytes, &logEntry)
+	if err != nil {
+		return nil, fmt.Errorf("could not unmarshal the logEntry from file: %s, err: %v", rekorLogFilePath, err)
+	}
+
+	if len(logEntry) != 1 {
+		return nil, fmt.Errorf("For transparent release, Rekor log entries should contain exactly one UUID: %v", logEntry)
+	}
+
+	var logEntryAnon models.LogEntryAnon
+	// set logEntryAnon to the only value in LogEntry (which is a map)
+	for _, anon := range logEntry {
+		logEntryAnon = anon
+	}
+	return &logEntryAnon, nil
+}
+
+func getRekordEntryFromAnon(logEntryAnon models.LogEntryAnon) (*rekord.V001Entry, error) {
+	bodyString, ok := logEntryAnon.Body.(string)
+	if !ok {
+		return nil, fmt.Errorf("could not coerce LogEntryAnon into string. LogEntryAnon: %v", logEntryAnon)
+	}
+
+	bodyDecoded, err := base64.StdEncoding.DecodeString(bodyString)
+	if err != nil {
+		return nil, fmt.Errorf("could not decode body from base64 %v: %v", logEntryAnon, err)
+	}
+
+	proposedEntry, err := models.UnmarshalProposedEntry(bytes.NewReader(bodyDecoded), runtime.JSONConsumer())
+	if err != nil {
+		return nil, fmt.Errorf("could not unmarshal proposed entry from body: %v, %s", bodyDecoded, err)
+	}
+
+	entryImpl, err := types.NewEntry(proposedEntry)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert ProposedEntry into NewEntry: %v, %s", proposedEntry, err)
+	}
+	// rekord
+	rekordEntry, ok := entryImpl.(*rekord.V001Entry)
+	if !ok {
+		return nil, fmt.Errorf("could not convert NewEntry into rekord. NewEntry: %v,", entryImpl)
+	}
+	return rekordEntry, nil
```

Comment:
Maybe we can do this when we know we need it as described in this newly created issue: #74 ?

I'm confused why we would expect other types. Since these are rekor logs of endorsement files which are specific to our software, can't we just choose which one format we use in all situations? Why do we need this flexibility?

Reply:
If we eventually want to use `hashedRekord` instead of `rekord` we could just change the one format we use in this code.

Reply:
OK I split.